### PR TITLE
[d3d11] A few optimizations of shaders and fixes.

### PR DESCRIPTION
**Pull request by afedchin**
@Paxxi, please review.

### CODE REVIEW

**Comment by fritsch** [comment]
For the tiny optimization - a one sentence summary would have been nice, so that not so educated reviewers (like me in that case ;-)) get an idea of what you optimize.

**Comment by afedchin** [comment]
sounds reasonable. Updated commits.

File: xbmc/cores/VideoRenderers/VideoShaders/WinVideoFilter.cpp [hunk 1]
```diff
@@ -409,8 +406,8 @@ void CYUV2RGBShader::SetShaderParameters(YUVBuffer* YUVbuf)
   {
     D3D11_TEXTURE2D_DESC rtDescr = {};
     rtTexture->GetDesc(&rtDescr);
-    viewPortWidth = rtDescr.Width;
-    viewPortHeight = rtDescr.Height;
+    viewPortWidth = (float)rtDescr.Width;
```
Comment (Paxxi) [inline xbmc/cores/VideoRenderers/VideoShaders/WinVideoFilter.cpp]: Use static_cast

File: xbmc/cores/VideoRenderers/VideoShaders/WinVideoFilter.cpp [hunk 2]
```diff
@@ -691,9 +685,14 @@ void CConvolutionShaderSeparable::Render(CD3DTexture &sourceTexture,
...
   PrepareParameters(sourceWidth, sourceHeight, destWidth, destHeight, sourceRect, destRect);
-  float texSteps1[] = { 1.0f/(float)sourceWidth, 1.0f/(float)sourceHeight};
-  float texSteps2[] = { 1.0f/(float)destWidth, 1.0f/(float)(sourceHeight)};
-  SetShaderParameters(sourceTexture, &texSteps1[0], ARRAY_SIZE(texSteps1), &texSteps2[0], ARRAY_SIZE(texSteps2));
+  float texSteps[] = 
+  { 
+    1.0f / (float)sourceWidth, 
+    1.0f / (float)sourceHeight, 
+    1.0f / (float)destWidth, 
+    1.0f / (float)sourceHeight
```
Comment (Paxxi) [inline xbmc/cores/VideoRenderers/VideoShaders/WinVideoFilter.cpp]: Same here

**Comment by Paxxi** [comment]
Other than the nitpicks about static_cast everywhere I can't review this, I don't understand any of that code :)

**Comment by afedchin** [comment]
@Paxxi, thanks. 
jenkins build this please

**Pull request merged**